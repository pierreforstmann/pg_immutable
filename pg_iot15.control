# pg_iot15 extension
comment = 'insert-only table AM '
default_version = '0.0.1'
module_pathname = '$libdir/pg_iot15'
relocatable = true
