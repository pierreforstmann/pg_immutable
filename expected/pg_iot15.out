CREATE EXTENSION pg_iot15;
CREATE TABLE pg_iot15_tab (a int) USING pg_iot15;
SELECT * FROM pg_iot15_tab;
 a 
---
(0 rows)

INSERT INTO pg_iot15_tab VALUES (1);
SELECT * FROM pg_iot15_tab;
 a 
---
 1
(1 row)

UPDATE pg_iot15_tab SET a = 0 WHERE a = 1;
ERROR:  pg_iot15: iot15_tuple_update: cannot UPDATE an iot15 table
SELECT * FROM pg_iot15_tab;
 a 
---
 1
(1 row)

DELETE FROM pg_iot15_tab WHERE a = 1;
ERROR:  pg_iot15: iot15_tuple_delete: cannot DELETE from an iot15 table
SELECT * FROM pg_iot15_tab;
 a 
---
 1
(1 row)

--
create table t(c int) using pg_iot15;
create index i on t(c);
insert into t select generate_series(1, 1000000);
delete from t;
ERROR:  pg_iot15: iot15_tuple_delete: cannot DELETE from an iot15 table
update t set c=3 where c=999999;
ERROR:  pg_iot15: iot15_tuple_update: cannot UPDATE an iot15 table
select * from t where c = 555555;
   c    
--------
 555555
(1 row)

-- ALTER TABLE SET ACCESS METHOD
ALTER TABLE pg_iot15_tab SET ACCESS METHOD heap;
ERROR:  Cannot change access method for iot15 am
SELECT * FROM pg_iot15_tab;
 a 
---
 1
(1 row)

update pg_iot15_tab set a=1;
ERROR:  pg_iot15: iot15_tuple_update: cannot UPDATE an iot15 table
-- Clean up
DROP TABLE pg_iot15_tab;
